internal/ls/diagnostics.go

```diff
@@ -13,17 +13,14 @@ import (
 func (l *LanguageService) ProvideDiagnostics(ctx context.Context, uri lsproto.DocumentUri) (lsproto.DocumentDiagnosticResponse, error) {
 	program, file := l.getProgramAndFile(uri)
 
-	diagnostics := make([][]*ast.Diagnostic, 0, 3)
-	if syntaxDiagnostics := program.GetSyntacticDiagnostics(ctx, file); len(syntaxDiagnostics) != 0 {
-		diagnostics = append(diagnostics, syntaxDiagnostics)
-	} else {
-		diagnostics = append(diagnostics, program.GetSemanticDiagnostics(ctx, file))
-		// !!! user preference for suggestion diagnostics; keep only unnecessary/deprecated?
-		// See: https://github.com/microsoft/vscode/blob/3dbc74129aaae102e5cb485b958fa5360e8d3e7a/extensions/typescript-language-features/src/languageFeatures/diagnostics.ts#L114
-		diagnostics = append(diagnostics, program.GetSuggestionDiagnostics(ctx, file))
-		if program.Options().GetEmitDeclarations() {
-			diagnostics = append(diagnostics, program.GetDeclarationDiagnostics(ctx, file))
-		}
+	diagnostics := make([][]*ast.Diagnostic, 0, 4)
```

Comment:
I had thought this code was correct originally; I guess you're changing it to match that VS Code on the client side requested syntax and semantic in sequence? (I've been trying to find equivalent code but I think this is just somewhere that differs in general with Strada given the fine-grained-ness of tsserver.)

Reply:
I simply noticed a behavioral difference, with semantic error red squiggles appearing and disappearing depending on whether there are syntax errors as you're typing. LSP doesn't distinguish between syntactic/semantic so we need to return all of them to get consistent behavior.